runtime/runtime.go

```diff
@@ -487,36 +482,64 @@ func (rt *Runtime) PrepareTransaction(tx *tx.Transaction) (*TransactionExecutor,
 			finalized = true
 
 			receipt := &Tx.Receipt{
-				Type:     tx.Type(),
+				Type:     trx.Type(),
 				Reverted: reverted,
 				Outputs:  txOutputs,
-				GasUsed:  tx.Gas() - leftOverGas,
+				GasUsed:  trx.Gas() - leftOverGas,
 				GasPayer: payer,
 			}
-
-			receipt.Paid = new(big.Int).Mul(new(big.Int).SetUint64(receipt.GasUsed), gasPrice)
+			receipt.Paid = new(big.Int).Mul(new(big.Int).SetUint64(receipt.GasUsed), effectiveGasPrice)
 
 			if err := returnGas(leftOverGas); err != nil {
 				return nil, err
 			}
 
-			// reward
-			rewardRatio, err := builtin.Params.Native(rt.state).Get(thor.KeyRewardRatio)
-			if err != nil {
-				return nil, err
-			}
-			provedWork, err := tx.ProvedWork(rt.ctx.Number-1, rt.chain.GetBlockID)
-			if err != nil {
-				return nil, err
+			if rt.ctx.Number < rt.forkConfig.GALACTICA {
```

Comment:
While this removes logic to the fork package, it raises the difficulty to unit test it.